lib/src/local_working_copy.rs

```diff
@@ -1021,23 +1021,27 @@ impl TreeState {
     ) -> Result<FsmonitorMatcher, SnapshotError> {
         let (watchman_clock, changed_files) = match fsmonitor_kind {
             None => (None, None),
+            Some(FsmonitorKind::None) => (None, None),
             Some(FsmonitorKind::Test { changed_files }) => (None, Some(changed_files)),
-            #[cfg(feature = "watchman")]
-            Some(FsmonitorKind::Watchman) => match self.query_watchman() {
-                Ok((watchman_clock, changed_files)) => (Some(watchman_clock.into()), changed_files),
-                Err(err) => {
-                    tracing::warn!(?err, "Failed to query filesystem monitor");
-                    (None, None)
-                }
-            },
-            #[cfg(not(feature = "watchman"))]
             Some(FsmonitorKind::Watchman) => {
-                return Err(SnapshotError::Other {
-                    message: "Failed to query the filesystem monitor".to_string(),
-                    err: "Cannot query Watchman because jj was not compiled with the `watchman` \
-                          feature (consider disabling `core.fsmonitor`)"
-                        .into(),
-                });
+                if cfg!(feature = "watchman") {
+                    match self.query_watchman() {
```

Comment:
The reason this was originally written this way is because `query_watchman` didn't exist/wouldn't compile without the `watchman` feature enabled. Can you make sure that this still compiles both with and without the `watchman` feature? (Not sure if it's checked in CI.)

Reply:
Ah, you're right, I thought `cfg!` would guard it appropriately but apparently not. I think this will break the build. I'll just drop this.